packages/platforms/react-native/android/src/main/java/com/bugsnag/android/performance/NativeBugsnagPerformanceImpl.java

```diff
@@ -74,6 +81,101 @@ public void requestEntropyAsync(Promise promise) {
     promise.resolve(requestEntropy());
   }
 
+  public WritableMap getNativeConstants() {
+    WritableMap map = Arguments.createMap();
+    map.putString("CacheDir", this.reactContext.getCacheDir().getAbsolutePath());
+    map.putString("DocumentDir", this.reactContext.getFilesDir().getAbsolutePath());
+
+    return map;
+  }
+
+  public void exists(String path, Promise promise) {
+    try {
+      boolean result = new File(path).exists();
+      if (result) {
+        promise.resolve(result);
+      } else {
+        promise.reject(new Exception("File does not exist"));
+      }
+    } catch(Exception e) {
+      promise.reject(e);
+    }
+  }
+
+  public void isDir(String path, Promise promise) {
+    try {
+      boolean result = new File(path).isDirectory();
+      if (result) {
+        promise.resolve(result);
+      } else {
+        promise.reject(new Exception("Path is not a directory"));
+      }
+    } catch(Exception e) {
+      promise.reject(e);
+    }
+  }
+
+  public void ls(String path, Promise promise) {
+    try {
+      promise.resolve(new File(path).list());
+    } catch(Exception e) {
+      promise.reject(e);
+    }
+  }
+
+  public void mkdir(String path, Promise promise) {
+    try {
+      boolean result = new File(path).mkdir();
+      if (result) {
+        promise.resolve(path);
+      } else {
+        promise.reject(new Exception("Failed to create directory"));
+      }
+    } catch(Exception e) {
+      promise.reject(e);
+    }
+  }
+
+  public void readFile(String path, String encoding, Promise promise) {
+    try {
+      StringBuilder fileContent = new StringBuilder();
+      InputStreamReader isr = new InputStreamReader(new FileInputStream(path), encoding);
+      BufferedReader fileReader = new BufferedReader(isr);
+      String dataLine = "";
+      while ((dataLine = fileReader.readLine()) != null) {
+        fileContent.append(dataLine);
+      }
+      fileReader.close();
```

Comment:
```suggestion
    File file = new File(path);
    StringBuilder fileContent = new StringBuilder((int) file.length());
    try(
      FileInputStream fin = new FileInputStream(file);
      InputStreamReader isr = new InputStreamReader(fin, encoding);
    ) {
      char[] buffer = new char[4096];
      int charsRead = 0;
      while ((charsRead = isr.read(buffer)) != -1) {
        fileContent.append(buffer, 0, charsRead);
      }
```

I think using a fixed sized buffer of characters and pre-allocating the `StringBuilder` is going to be better than reading the lines one-by-one.

Also using `try-with-resources` will handle close operations for us.